src/FolderConfig.cpp

```diff
@@ -0,0 +1,94 @@
+#include "FolderConfig.h"
+#include <QFile>
+#include <QDebug>
+#include<QDir>
+
+namespace QtDBMigrationNS {
+FolderConfig::FolderConfig(const QString &path): AbstractConfig(path)
+{
+}
+
+bool FolderConfig::load()
+{
+    QDir dir(m_path);
+    QStringList nameFilter("*.sql");
+    QStringList sqlFiles = dir.entryList(nameFilter);
+    int totalFiles = sqlFiles.count();
+
+    if (!dir.exists() || totalFiles < 1) {
+        return false;
+    }
+
+    for(auto filename : sqlFiles) {
+        bool migrationsUp = true;
+        QString filePath = dir.filePath(filename);
+        QFile file(filePath);
+        if (!file.exists()) {
+            return false;
+        }
+
+        QFileInfo fileInfo(file.fileName());
+        QRegExp rx("^\\d{4}\\.sql$");
+        if (!rx.exactMatch(fileInfo.fileName())) {
+            continue;
+        }
+
+        if(!file.open(QFile::ReadOnly | QIODevice::Text)) {
+            return false;
+        }
+
+        QTextStream in(&file);
+        QString line = in.readLine();
+        QStringList applySql;
+        QStringList revertSql;
+        while (!line.isNull())
+        {
+            if (line.size()==0)
+            {
+                line = in.readLine();
+                continue;
+            }
+
+            if (line.contains("-- +"))
+            {
+                if(line.contains("Up"))
+                {
+                    migrationsUp = true;
+                }
+                else if (line.contains("Down"))
+                {
+                    migrationsUp = false;
+                }
+                line = in.readLine();
+                continue;
+            }
+
+            if (migrationsUp)
+            {
+                applySql.append(line);
```

Comment:
Will this survive queries like:
```sql
alter table contacts
  modify col1 int,
  modify col2 varchar,
  add col2 int;
```

Try adding a test first, and fix later ;).

Reply:
Hm, testing this on SQLite is impossible, since adding and dropping columns does not work as easy. But if it is the format of the queries then I will just drop the last statement.